tests/foreman/api/test_http_proxy.py

```diff
@@ -0,0 +1,142 @@
+"""Tests for http-proxy component.
+
+:Requirement: HttpProxy
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Repositories
+
+:Assignee: jpathan
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+import pytest
+from fauxfactory import gen_string
+
+from robottelo import constants
+from robottelo.api.utils import enable_rhrepo_and_fetchid
+from robottelo.config import settings
+
+
+def create_http_proxy(sat, org, proxy_type):
+    """
+    Creates HTTP proxy.
+
+    :param str sat: Satellite to use.
+    :param str org: Organization
+    :param str proxy_type: 'auth_http_proxy' or 'unauth_http_proxy' http proxy.
+    """
+    if proxy_type == 'unauth_http_proxy':
+        return sat.api.HTTPProxy(
+            name=gen_string('alpha', 15),
+            url=settings.http_proxy.un_auth_proxy_url,
+            organization=[org.id],
+        ).create()
+    if proxy_type == 'auth_http_proxy':
+        return sat.api.HTTPProxy(
+            name=gen_string('alpha', 15),
+            url=settings.http_proxy.auth_proxy_url,
+            username=settings.http_proxy.username,
+            password=settings.http_proxy.password,
+            organization=[org.id],
+        ).create()
+
+
+@pytest.fixture(scope='function')
+def function_http_proxy(request, module_manifest_org, target_sat):
+    """Create a new HTTP proxy and set related settings based on proxy"""
+    http_proxy = create_http_proxy(target_sat, module_manifest_org, request.param)
```

Comment:
Since the calls are the same, just with different kwargs, *consider* not having a separate function for the 'create()` call, and just do it here. Create a kwargs dict with if/else blocks, and then expand those kwargs into a single call to `sat.api.HTTPProxy.create()`